Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -859,10 +859,12 @@ public bool UnregisterPointer(IMixedRealityPointer pointer)
         {
             using (UnregisterPointerPerfMarker.Auto())
             {
-                Debug.Assert(pointer.PointerId != 0, $"{pointer} does not have a valid pointer id!");
+                if (pointer.PointerId != 0)
```

Comment:
Could you elaborate on this change?

Reply:
Definitely! Basically, the interpolated string causes an allocation every time the Debug.Assert is called, even if the case is true and we don't hit the assert and log. It's kinda a roundabout way of fixing this, as what's the point of the assert if we check the failure case before we even call it...but I haven't been able to come up with a better way :/ open to suggestions though!